src/ui/public/agg_types/buckets/geo_hash.js

```diff
@@ -114,29 +108,29 @@ export const geoHashBucketAgg = new BucketAggType({
       controller: function () {
       },
       write: function (aggConfig, output) {
-        const vis = aggConfig.vis;
-        const currZoom = getMapZoom(vis);
+        const currZoom = aggConfig.params.mapZoom;
         const autoPrecisionVal = zoomPrecision[currZoom];
-        output.params.precision = aggConfig.params.autoPrecision ? autoPrecisionVal : getPrecision(aggConfig.params.precision);
+        if (autoPrecisionVal) {
+          output.params.precision = aggConfig.params.autoPrecision ?
+            autoPrecisionVal : getPrecision(aggConfig.params.precision);
+        }
       }
     }
   ],
   getRequestAggs: function (agg) {
     const aggs = [];
+    const { vis, params } = agg;
 
-    if (agg.params.isFilteredByCollar && agg.getField()) {
-      const vis = agg.vis;
-      const mapBounds = vis.sessionState.mapBounds;
-      const mapZoom = getMapZoom(vis);
+    if (params.isFilteredByCollar && agg.getField()) {
+      const { mapBounds, mapZoom } = params;
       if (mapBounds) {
-        const lastMapCollar = vis.sessionState.mapCollar;
         let mapCollar;
-        if (!lastMapCollar || lastMapCollar.zoom !== mapZoom || isOutsideCollar(mapBounds, lastMapCollar)) {
+        if (!agg.lastMapCollar || agg.lastMapCollar.zoom !== mapZoom || isOutsideCollar(mapBounds, agg.lastMapCollar)) {
```

Comment:
Why are `mapBounds` and `mapZoom` stored in `agg.params` but `lastMapCollar` is stored directly under `agg`?

Reply:
because `lastMapCollar` is not a configuration to the aggregation